src/pages/map/map.ts

```diff
@@ -78,4 +114,57 @@ export class MapPage {
     this.layers.push(new Marker(location.id, location.geometry.coordinates, { icon: defIcon }));
   }
 
-}
\ No newline at end of file
+  /**
+   * Center the map on the user's current position if in Onex
+   * (otherwise remain at the default position).
+   */
+  private autoCenterMap() {
+    this.getCurrentPosition().then(currentPosition => {
+      if (turf.inside(currentPosition, ONEX_BBOX)) {
+        this.centerMap(currentPosition);
+      }
+    }).catch(() => { /* do nothing */ });
+  }
+
+  /**
+   * Gets the user's current position, displaying a message on the map while it
+   * is being determined.
+   *
+   * @param {function} [action] - An optional action to execute once the position is available.
+   * @returns {Promise<UserPosition>} A GeoJSON point feature.
+   */
+  private getCurrentPosition(): Promise<UserPosition> {
+
+    this.geolocationInProgress = true;
+    this.mapMessage = this.translateService.instant('pages.map.geolocation');
+
+    return this.geolocation
+      .getCurrentPosition()
+      .then(result => {
+        this.setGeolocationDone();
+        return turf.point([ result.coords.longitude, result.coords.latitude ]);
+      }, err => {
+        this.setGeolocationDone();
+        Print.warn('Could not get user position', err);
+        return Promise.reject(err);
```

Comment:
Should be a message shown to the user telling them that their location couldn't be found.